src/openrc-run/openrc-run.c

```diff
@@ -448,9 +451,18 @@ svc_exec(const char *arg1, const char *arg2)
 				write_prefix(buffer, bytes, &prefixed);
 			}
 
-			/* Only SIGCHLD signals come down this pipe */
-			if (fd[0].revents & (POLLIN | POLLHUP))
+			/* Only SIGCHLD signals come down this pipe,
+			 * but with rc_parallel is might be another process */
```

Comment:
I think this comment should be added above because it's misleading in here - since with the current patch - if it's another process then nothing will be written on the pipe.